File: TDDMicroExercises.ProposedSolution/CSharp/TelemetrySystem.SomeDependencies/TelemetryClientClient.cs
namespace TDDMicroExercises.OneSolution.TelemetrySystem.SomeDependencies
{
    public class TelemetryClientClient
    {
		// A class with the only goal of simulating a dependency on TelemetryClient
		// that has impact on the refactoring.

		public TelemetryClientClient()
        {
            var tc = new TelemetryClientConnection();
            if (!tc.OnlineStatus)
                tc.Connect("a connection string");

            var tdata = new TelemetryDataChannel();

            tdata.Send("some message");

            var response = tdata.Receive();

            tc.Disconnect();

        }
    }
}

File: TDDMicroExercises/Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/somedependencies/TelemetryClientClient.java
package tddmicroexercises.telemetrysystem.somedependencies;

import tddmicroexercises.telemetrysystem.TelemetryClient;

public class TelemetryClientClient
{
    // A class with the only goal of simulating a dependency on TelemetryClient
    // that has impact on the refactoring.

    public TelemetryClientClient()
    {
        TelemetryClient tc = new TelemetryClient();
        if (!tc.getOnlineStatus())
            tc.connect("a connection string");

        tc.send("some message");

        String response = tc.receive();

        tc.disconnect();

    }
}
